### PR TITLE
Jetpack Cloud: Add History Items

### PR DESCRIPTION
**Pull request by enejb**
#### Changes proposed in this Pull Request
This PR adds Historic Threats that are fixed or ignored by using the new QueryScanHistory 
component. 

I think a lot of this data that comes from the api still needs clean up. 

After:
Placeholder:
Screen Shot 2020-04-17 at 2 09 53 PM

With data from the api.
Screen Shot 2020-04-17 at 2 07 22 PM

#### Testing instructions

Load This PR and visit 
* http://jetpack.cloud.localhost:3000/scan/history/example.com
* Notice that the treats info shows up as expected.

### CODE REVIEW

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx [hunk 1]
```diff
@@ -25,61 +28,216 @@ class ScanHistoryItem extends Component {
...
+			case 'theme':
+				return translate( 'Vulnerable Theme {{themeSlug/}} (version {{version/}})', {
+					components: {
+						themeSlug: <span className="threat-item__alert-slug">{ extension.slug }</span>,
+						version: <span className="threat-item__alert-version">{ extension.version }</span>,
+					},
+				} );
+
+			case 'database':
+				return translate( 'Database %(threatCount)d threat', 'Database %(threatCount)d threats', {
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/scan-history-item/index.jsx:164]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Jetpack identified %(threatCount)d threat in your database.', 'Jetpack identified %(threatCount)d threats in your database.' )` *ES Score: 7*

**Comment by enejb** [comment]
There is defenetly some overlap with this PR https://github.com/Automattic/wp-calypso/pull/41216

File: client/landing/jetpack-cloud/sections/scan/history/index.jsx [hunk 2]
```diff
@@ -11,12 +11,13 @@ import page from 'page';
...
 import DocumentHead from 'components/data/document-head';
 import QueryJetpackScanHistory from 'components/data/query-jetpack-scan-history';
-import ScanHistoryItem from '../../../components/scan-history-item';
+import ScanHistoryItem from 'landing/jetpack-cloud/components/scan-history-item';
 import SimplifiedSegmentedControl from 'components/segmented-control/simplified';
 import { getSelectedSiteSlug, getSelectedSiteId } from 'state/ui/selectors';
 import Main from 'components/main';
 import SidebarNavigation from 'my-sites/sidebar-navigation';
-
+import isRequestingJetpackScanHistory from 'state/selectors/is-requesting-jetpack-scan-history';
```
Comment (rcanepa) [inline client/landing/jetpack-cloud/sections/scan/history/index.jsx:19]: Should we rename this selector to something like `isRequestingSiteScanHistory`?

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx [hunk 3]
```diff
@@ -25,61 +28,216 @@ class ScanHistoryItem extends Component {
...
+										},
+									} ) }
+								</p>
+								<pre className="threat-description__alert-filename">{ threat.filename }</pre>
+							</>
+						) : (
+							<p className="threat-description__section-text">{ threat.signature }</p>
+						) }
+						{ threat.context && <MarkedLines context={ threat.context } /> }
+						{ threat.diff && <DiffViewer diff={ threat.diff } /> }
```
Comment (rcanepa) [inline client/landing/jetpack-cloud/components/scan-history-item/index.jsx:206]: I think we should add a `margin-bottom` of 16 or 24px to the `DiffViewer` component. It seems to be too close to the title that is just below it.
image

**Comment by ChaosExAnima** [comment]
@rcanepa - Thinking it would be good to use `ThreatItem` between these two pages? In the end we're using these parsing functions for display of threat details, whether current or historical.

**Comment by rcanepa** [comment]
> @rcanepa - Thinking it would be good to use `ThreatItem` between these two pages? In the end we're using these parsing functions for display of threat details, whether current or historical.

Yes, I believe doing that makes a lot of sense.

**Comment by rcanepa** [comment]
I pushed some important changes to this PR:

- Extracted the logic used to generate the header present in both the ThreatItem and ScanHistoryItem components into a component called ThreatItemHeader. Thanks to this, I could remove a lot of duplication from the ScanHistoryItem.
- Transformed the ScanHistoryItem main `index.jsx` to `index.tsx` (TS).
- Updated the ThreatDescription component README file.
- In the ThreatDescription component, I renamed the `action` prop  to `status` to use the same name the API response has. I updated this in our Threat type as well.
- Removed the `details` prop of the ThreatDescription component, because it was no longer useful.
- Updated the ThreatItem README file.

image
image

**Comment by rcanepa** [comment]
Hey @keoshi ! What do you think about the positioning of the "Ignore threat" button, in the last screenshot, in relation to the rest of the content ? Because we are limiting the max-width of the `.main` section, it seems the space for the content is too small now. Should move the button on top of the content and let the content use the whole width of the card?

**Comment by keoshi** [comment]
Sorry @rcanepa — doing my support rotation and missed this.

> Should move the button on top of the content and let the content use the whole width of the card?

Yes, please! But also, I'd try moving the button to the bottom so customers read the context of a threat before taking action. Does that make sense?

**Comment by rcanepa** [comment]
No problem @keoshi ! I forgot you were doing you support rotation.

> Does that make sense?

Absolutely, we will take care of doing that!

**Pull request merged**